add Budget class, budget and showbudget command

austinsantoso opened the pull request:
add budget class, with save and load functionalities. added budget and showbudget commands

choongyx commented on src/test/java/seedu/address/logic/commands/BudgetCommandTest.java:
```diff
@@ -0,0 +1,29 @@
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    void execute_setNewBudget_success() {
```
public void --> to remain consistent with other tests and for code quality check

choongyx commented on src/test/java/seedu/address/logic/commands/ShowBudgetCommandTest.java:
```diff
@@ -0,0 +1,26 @@
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    void execute_displayBudget_success() {
```
public void --> to remain consistent with other tests and for code quality check

choongyx commented on src/main/java/seedu/address/model/budget/Budget.java:
```diff
@@ -0,0 +1,78 @@
+ */
+public class Budget {
+
+    public static final String MESSAGE_CONSTRAINTS = "Budget values should be positive number";
```
Are we allowing 2dp for budget (like Cost) or budget can use Cost too?

choongyx commented:
#30

austinsantoso committed:
add Budget class, budget and showbudget command

add budget class, with save and load functionalities. added budget and show budget commands
budget code quality

Nanosync commented on src/main/java/seedu/address/logic/commands/ShowBudgetCommand.java:
```diff
@@ -0,0 +1,18 @@
+import seedu.address.model.Model;
+
+/**
+ * Displays the current SPending
```
Budget

Nanosync commented on src/test/java/seedu/address/logic/parser/BudgetCommandParserTest.java:
```diff
@@ -0,0 +1,27 @@
+
+    @Test
+    public void parse_invalidArgs_throwsParseException() {
+        assertParseFailure(parser, "", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
```
add a negative value argument for test case

austinsantoso committed:
add test cases

add test cases for budget
